includes/subscriptions.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+class Migrator_CLI_Subscriptions {
+	public function import( $assoc_args ) {
+
+		try {
+			Migrator_CLI_Utils::health_check();
+
+			if ( !isset( $assoc_args['subscriptions_export_file'] ) || !isset( $assoc_args['orders_export_file'] ) ) {
+				WP_CLI::line( WP_CLI::colorize( '%RExport Files not provided. Go to Skio dashboard > Export and export both subscriptions and orders. Then pass the file path to --subscriptions_export_file and --orders_export_file args' ) );
+				return;
+			}
+
+			$skio_orders = $this->get_data_from_file( $assoc_args['orders_export_file'] );
+			$skio_subscriptions = $this->get_data_from_file( $assoc_args['subscriptions_export_file'] );
+
+			Migrator_CLI_Utils::disable_sequential_orders();
+
+			WP_CLI::line( 'Adding Subscription ids to orders' );
+			$this->add_subscription_id_to_orders( $skio_orders );
+
+			WP_CLI::line( 'Creating Subscriptions' );
+			$this->create_or_update_subscriptions( $skio_subscriptions );
+
+			Migrator_CLI_Utils::enable_sequential_orders();
+		} catch ( \Exception $e ) {
+			WP_CLI::line( WP_CLI::colorize( ' %rError: ' . $e->getMessage() ) );
+		}
+
+		WP_CLI::line( WP_CLI::colorize( '%GDone%n' ) );
+	}
+
+	// Clones the line item and adds it to the subscription.
+	private function clone_item_to_subscription( $item, $subscription ) {
+		$new_item = clone $item;
+		$new_item->set_id( 0 );
+		$new_item->set_order_id( $subscription->get_id() );
+		$new_item->save();
+
+		$subscription->add_item( $new_item );
+	}
+
+	private function get_data_from_file( $file )
+	{
+		if ( ! is_file( $file ) ) {
```

Comment:
I see the `File not found` error, even if I provide the correct absolute paths. This may be because we are using  `is_file`, that does not work with absolute paths. 

We may need to use something like this: 

```suggestion
		private function get_data_from_file( $file )
	{
		$file_contents = file_get_contents( $file );

		if ( $file_contents === false ) {
			WP_CLI::line( WP_CLI::colorize( '%RFile not found: ' . $file ) );
			die();
		}

		return json_decode( $file_contents, true);
	}
	
	
```

Reply:
If i change this and test, the duplication of items still exists when I re-run the command.

Reply:
Duplication problem fixed with 1ce0178188ef44a5dbe92508cc3e79a4ca3760e8

Reply:
I tried running the command as follow:

`wp migrator skio_subscriptions --subscriptions_export_file=/Users/leonardoalbuquerque/Local/taika/app/public/wp-content/plugins/migrator-cli/.tmp/subscriptions.json  --orders_export_file=/Users/leonardoalbuquerque/Local/taika/app/public/wp-content/plugins/migrator-cli/.tmp/orders.json`

And it found the file. 

What did not work was to run it like this:

`wp migrator skio_subscriptions --subscriptions_export_file=../.tmp/subscriptions.json  --orders_export_file=../.tmp/orders.json`

Reply:
Ah it did not work for me since I used URL of the localhost. and is_file does not work with URL paths. I will try again with the local path. We may need to add this in the Readme.

Reply:
@leonardola  The duplication issue seems to be resolved. The command also works correctly with the full local path.    However, for these steps I do not see the order addresses changing :

```
* Change the shipping and billing address in the newest order
* Run wp migrator skio_subscriptions --subscriptions_export_file=<file_path>  --orders_export_file=<file_path> again
* Make sure the subscription got updated
```

If it works fine at your end, it could be very likely my local setup.

Reply:
> However, for these steps I do not see the order addresses changing for these testing steps shared by you:

Sometimes the wrong order was being used to create/update the subscription. Fixed with: 872b8f1d3e64336312061df228a55415bfa4dd50